source/opt/scalar_replacement_pass.cpp

```diff
@@ -335,8 +335,18 @@ uint32_t ScalarReplacementPass::GetOrCreatePointerType(uint32_t id) {
   auto iter = pointee_to_pointer_.find(id);
   if (iter != pointee_to_pointer_.end()) return iter->second;
 
-  // TODO(alanbaker): Make the type manager useful and then replace this code.
+  opt::analysis::Type* pointeeTy = context()->get_type_mgr()->GetType(id);
+  opt::analysis::Pointer pointerTy(pointeeTy, SpvStorageClassFunction);
```

Comment:
Could this pattern be encoded in the type manager as a single function call?  It happens in a couple of places.  It could return a std::tie<pointeeTy, pointerTy> or an std::pair.  Not a biggie, though.

Reply:
Ok